src/role_carry.js

```diff
@@ -37,8 +37,8 @@ roles.carry.preMove = function(creep, directions) {
       var source = creep.room.memory.position.creep[targetId];
       // TODO better the position from the room memory
       if (source !== null) {
-        let returnCode = creep.moveTo(source.pos);
-        if (creep.pos.getRangeTo(source.pos) > 1) {
+        let returnCode = creep.moveTo(targetId);
+        if (creep.pos.getRangeTo(targetId) > 1) {
```

Comment:
Use Game.getObjectById(targetID) I dont believe moveTo can take Id numbers directly, either position or object.
Bonus points if we can use saved paths here?